drivers/include/lis3mdl.h

```diff
@@ -0,0 +1,168 @@
+ */
+typedef enum {
+    LIS3MDL_Z_MODE_LOW    = 0x00,    /** Low-power mode (default)    */
+    LIS3MDL_Z_MODE_MEDIUM = 0x04,    /** Medium-performance mode     */
+    LIS3MDL_Z_MODE_HIGH   = 0x08,    /** High-performance mode       */
+    LIS3MDL_Z_MODE_ULTRA  = 0x0C,    /** Ultra-High-performance mode */
+} lis3mdl_z_mode_t;
+
+/**
+ * @brief Output data rate [Hz] for LIS3MDL (CTRL_REG1)
+ */
+typedef enum {
+    LIS3MDL_ODR_0_625Hz = 0x00,      /**  0.625Hz (default) */
+    LIS3MDL_ODR_1_25Hz  = 0x04,      /**  1.250Hz           */
+    LIS3MDL_ODR_2_5Hz   = 0x08,      /**  5.000Hz           */
+    LIS3MDL_ODR_10Hz    = 0x10,      /** 10.000Hz           */
+    LIS3DML_ODR_20HZ    = 0x14,      /** 20.000Hz           */
+    LIS3DML_ODR_40HZ    = 0x18,      /** 40.000Hz           */
+    LIS3MDL_ODR_80HZ    = 0x1C,      /** 80.000Hz           */
+} lis3mdl_odr_t;
+
+/**
+ * @brief Scale [gauss] for LIS3MDL (CTRL_REG2)
+ */
+typedef enum {
+    LIS3MDL_SCALE_4G  = 0x00,        /** +-  4 gauss (default) */
+    LIS3MDL_SCALE_8G  = 0x20,        /** +-  8 gauss           */
+    LIS3MDL_SCALE_12G = 0x40,        /** +- 12 gauss           */
+    LIS3MDL_SCALE_16G = 0x60,        /** +- 16 gauss           */
+} lis3mdl_scale_t;
+
+/**
+ * @brief Operating modes
+ */
+typedef enum {
+    LIS3MDL_MD_CONT_CONV = 0x00,     /** Continous-conversion mode            */
+    LIS3MDL_MD_SNGL_CONV = 0x01,     /** Single-conversion mode 0.625Hz - 80Hz*/
+    LIS3MDL_MD_PDOW      = 0x11,     /** Power-down mode (default)            */
+} lis3mdl_md_t;
+
+/**
+ * @brief Initialize a new LIS3DML device.
+ *
+ * @param[in] dev           device descriptor of LIS3MDL
+ * @param[in] i2c           I2C device connected to
+ * @param[in] mag_address   I2C address of the magnometer
+ * @param[in] mag_xy_mode   power mode of x- & y-axis
+ * @param[in] mag_z_mode    power mode of z-axis
+ * @param[in] mag_odr       output data rate of magnometer
+ * @param[in] mag_scale     scale configuration of magnometer
+ *
+ * @return                   0 on success
+ * @return                  -1 on error
+ */
+int lis3mdl_init(lis3mdl_t *dev, i2c_t i2c, uint8_t mag_address,
+                 lis3mdl_xy_mode_t mag_xy_mode, lis3mdl_z_mode_t mag_z_mode,
+                 lis3mdl_odr_t mag_odr, lis3mdl_scale_t mag_scale,
+                 lis3mdl_md_t mag_op_mode);
+
+/**
+* @brief Reads the magnometer value of LIS3MDL.
+*
+* @param[in] dev    device descriptor of LIS3MDL
+* @param[in] data   measured magnetometer data
+*
+* @return            0 on success
+* @return           -1 on error
+*/
+int lis3mdl_read_mag(lis3mdl_t *dev, lis3mdl_3d_data_t *data);
+
+/**
+* @brief Reads the temperature value of LIS3MDL.
+*
+* @param[in] dev    device descriptor of LIS3MDL
+* @param[in] value  measured temperature
+*
+* @return            0 on success
+* @return           -1 on error
+*/
+int lis3mdl_read_temp(lis3mdl_t *dev, uint16_t *value);
```

Comment:
are you sure the temperature can only be positive? Also it would be nice, if you could state the physical unit and scaling of the value that is read (e.g. °C, °F, milidegree, etc.)